PLANNER-1732: Give Airbnb config highest priority

yurloc opened the pull request:
https://issues.jboss.org/browse/PLANNER-1732

For code style and project configuration consistency it is important to unify `.eslintrc.json` between Vehicle Routing and Employee Rostering.

Some time ago I have introduced the airbnb ESLint config (#251) but I've given it a low priority to avoid introducing big changes. Now's the time to take a step further and give it the top priority.

In the next step I'd like to eliminate most of the exceptions to the style (disabled rules).

Christopher-Chianelli commented on employee-rostering-frontend/cypress/plugins/index.js:
```diff
@@ -1,19 +1,3 @@
-/*
```
What happened to the license?

yurloc replied:
Because those were generated files - we're not the author => no need to license them. The license was only added to make ESLint happy. In my opinion it's better to disable ESLint completely for those folders (until we start adding code there) but keep the generated examples because:
- they're good examples, they might spark a new ideas when facing Cypress configuration issues,
- they would be generated on each Cypress run so we need to either gitignore them or check them in. The second option is better in my opinion.

Christopher-Chianelli replied:
We added lines there (`require('cypress-failed-log');`) and ```
  on('task', {
    failed: require('cypress-failed-log/src/failed')(),
  })
```

Christopher-Chianelli replied:
(those lines show actions when a cypress task fail in the log)

yurloc commented:
- Reverted `.eslintignore` => support and plugins are checked again.
- Added `cypress/**` under `devDependencies` of the `import/no-extraneous-dependencies` rule.
- Disabled `global-require` rule for cypress dir.

yurloc commented:
Jenkins retest this.

yurloc commented:
Added test coverage for the new default cases.

Christopher-Chianelli commented on employee-rostering-frontend/src/ui/Alerts.tsx:
```diff
@@ -45,10 +45,8 @@ const mapDispatchToProps: DispatchProps = {
-    }
+  if (component === AlertComponent.SERVER_SIDE_EXCEPTION_DIALOG) {
+    return <ServerSideExceptionDialog {...componentProps as unknown as ServerSideExceptionInfo} />;
   }
```
Is "undefined" a `React.ReactNode`? If not, I don't think it will pass a typecheck, unless typescript sees "SERVER_SIDE_EXCEPTION_DIALOG" is the only `AlertComponent`, in which TypeScript should complain about an if that is always true.

yurloc replied:
Good question. Yes it is.
```ts
    type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
```

yurloc replied:
Currently, there's only one `AlertComponent`:
```ts

export enum AlertComponent {
  SERVER_SIDE_EXCEPTION_DIALOG = 'SERVER_SIDE_EXCEPTION_DIALOG'
}
```
but it's still technically possible for the condition to be `false` in runtime.
